lesson3.md

```diff
@@ -0,0 +1,841 @@
+<!-- .slide: id="lesson3" -->
+
+# Basic Frontend - Spring 2020
+
+Lesson 3, Tuesday, 2020-09-01
+
+
+---
+
+
+### what we learned last time
+
+* Variables are pointers to values
+
+* We can use operators with variables just like with values
+
+* Use `<script src="xxx.js"></script>` to add a JavaScript file to your HTML webpage.
+
+* Use `//` to write comments.
+
+* Use `console.log(...)` to output values to the console.
+
+
+---
+
+
+### We had a HomeWork
+
+You have an online shop, which sells hats and t-shirts. One of the customers wants to buy 2 hats and 3 t-shirts.
+
+A hat costs 3.99€ and a t-shirt costs 9.99€. What is the total cost?
+
+Create all the variables needed, and output the final result to the console.
+
+
+---
+
+
+### Homework solution
+
+```js
+let hatPrice = 3.99;
+let tShirtPrice = 9.99;
+
+let hatAmount = 2;
+let tShirtAmount = 3;
+
+let totalHatPrice = hatAmount * hatPrice;
+let totalTShirtPrice = tShirtAmount * tShirtPrice;
+
+let totalCost = totalHatPrice + totalTShirtPrice;
+
+console.log(totalCost);
+```
+
+
+---
+
+
+### Recap,  What does it print?
+
+```js
+let name = "Sevtap";
+console.log("Hello " + name);
+```
+
+Answer: It prints "Hello Sevtap" <!-- .element: class="fragment" -->
+
+
+---
+
+
+### What does it print?
+
+```js
+let greeting = "Hello";
+let name = "Sevtap";
+console.log(greeting + " " + firstName);
+```
+
+Answer: Error, "firstName" is not a variable <!-- .element: class="fragment" -->
+
+
+---
+
+
+### What does it print?
+
+```js
+console.log("It is " + temperature + " degrees");
+let temperature = 22;
+```
+
+Answer: Error, "temperature" is not available <!-- .element: class="fragment" -->
```

Comment:
You use "no available" here, while you use "not a variable" before.
Chrome outputs "not defined" in both cases, while ff gives a more descriptive error in the second example and also "not defined" in the first one. 
Maybe it would be better to use the same messaging as the browser.

Reply:
I will stick with the output of chrome because that is what we are using in class

Reply:
Argh, seems indeed that in JavaScript, variables are not declared (sorry, I'm a C++ guy ;)) but "defined". So we should stick with "defined" for variables as well. Well spotted, @pektinasen  :)